presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroRowDecoder.java

```
@@ -55,7 +63,59 @@ private AvroColumnDecoder createColumnDecoder(DecoderColumnHandle columnHandle)
     @Override
     public Optional<Map<DecoderColumnHandle, FieldValueProvider>> decodeRow(byte[] data, Map<String, String> dataMap)
     {
-        GenericRecord avroRecord;
+        Optional<GenericRecord> candidate = decodeAvroDataFileFormat(data);
+        if (!candidate.isPresent()) {
+            candidate = decodeAvroSchemaRegistryFormat(data);
+        }
+
+        GenericRecord avroRecord = candidate
+                        .orElseThrow(() -> new PrestoException(GENERIC_INTERNAL_ERROR, "Decoding Avro record failed."));
+
+        return Optional.of(columnDecoders.entrySet().stream()
+                .collect(toImmutableMap(
+                        Map.Entry::getKey,
+                        entry -> entry.getValue().decodeField(avroRecord))));
+    }
+
+    private void closeQuietly(DataFileStream<GenericRecord> stream)
```

why not to close loudly? Why you swallow exception?